presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java

```diff
@@ -199,15 +207,72 @@ public void testCreatePartitionedTableAs(StorageFormat storageFormat)
         query(createTableAsSelect);
 
         // SELECT FROM TABLE
-        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName);
+        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName, TEST_TPCH_LINIETEM);
 
         // DROP TABLE
         query(format("DROP TABLE %s", tableName));
     }
 
-    private static void assertSelect(String query, String tableName)
+    @Requires(ImmutableLineItemTable.class)
+    @Test(groups = {STORAGE_FORMATS, BIG_QUERY})
+    public void testSelectFromPartitionedBzipTable()
     {
-        QueryResult expected = query(format(query, "tpch." + TPCH_SCHEMA + ".lineitem"));
+        String tableName = "storage_formats_test_select_partitioned_bzip";
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // The BZIP part of the table comes from the configs that are set during insert
+        String createTable = format(
+                "CREATE TABLE %s(" +
+                        "   l_orderkey      BIGINT," +
+                        "   l_partkey       BIGINT," +
+                        "   l_suppkey       BIGINT," +
+                        "   l_linenumber    INT," +
+                        "   l_quantity      DOUBLE," +
+                        "   l_extendedprice DOUBLE," +
+                        "   l_discount      DOUBLE," +
+                        "   l_tax           DOUBLE," +
+                        "   l_linestatus    VARCHAR(1)," +
+                        "   l_shipinstruct  VARCHAR(25)," +
+                        "   l_shipmode      VARCHAR(10)," +
+                        "   l_comment       VARCHAR(44)" +
+                        ") PARTITIONED BY (l_returnflag VARCHAR(1)) ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' STORED AS TEXTFILE",
+                tableName);
+        onHive().executeQuery(createTable);
+
+        try {
+            String insertInto = format(
+                    "INSERT OVERWRITE TABLE %s PARTITION(l_returnflag) " +
+                            "SELECT l_orderkey, l_partkey, l_suppkey, l_linenumber, l_quantity, l_extendedprice, l_discount, l_tax, " +
+                            "l_linestatus, l_shipinstruct, l_shipmode, l_comment, l_returnflag " +
+                            "FROM default.lineitem", tableName);
+            Statement statement = onHive().getConnection().createStatement();
+            setHiveConfigsForBzipInsert(statement);
+            statement.execute(insertInto);
+            statement.close();
+
+            assertSelect("select sum(l_tax), sum(l_discount), sum(length(l_returnflag)) from %s", tableName, "hive.default.lineitem");
+        }
+        catch (SQLException e) {
+            LOGGER.error("Exception in query:", e);
```

Comment:
Maybe you could mirror https://github.com/prestodb/presto/commit/9ec28091bd11c8cef9dc9163a0c945b8be4e651c here? Would be more consistent and you'd avoid this long "try-catch" block.